src/DIRAC/WorkloadManagementSystem/Agent/test/Test_Agent_JobAgent.py

```diff
@@ -606,8 +608,14 @@ def test_submitAndCheckJob(mocker, localCE, job, expectedResult1, expectedResult
     mocker.patch("DIRAC.WorkloadManagementSystem.Agent.JobAgent.JobAgent._sendFailoverRequest", return_value=S_OK())
     mocker.patch("DIRAC.Core.Security.X509Chain.X509Chain.dumpAllToString", return_value=S_OK())
     mocker.patch(
-        "DIRAC.Resources.Computing.SingularityComputingElement.SingularityComputingElement._SingularityComputingElement__hasSingularity",
-        return_value=False,
+        "DIRAC.Resources.Computing.SingularityComputingElement.SingularityComputingElement.submitJob",
+        return_value=S_ERROR("Failed to find singularity"),
```

Comment:
Looks like `Failed to find singularity` does not exist anymore:
```suggestion
        return_value=S_ERROR("Fail to run Singularity"),
```

Reply:
It doens't but the tests are written under the assumption it does